bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/WidgetsChangeListener.java

```diff
@@ -206,11 +205,12 @@ private Set<SitemapEvent> constructSitemapEvents(Item item, State state, List<Wi
                 events.addAll(constructSitemapEvents(item, state, itemUIRegistry.getChildren(grid)));
             }
 
-            boolean itemBelongsToWidget = w.getItem() != null && w.getItem().equals(item.getName());
+            boolean itemBelongsToWidget = w.getItem() != null && item.getName().equals(w.getItem());
             boolean skipWidget = !itemBelongsToWidget;
             // We skip the chart widgets having a refresh argument
             if (!skipWidget && w instanceof Chart chartWidget) {
-                skipWidget = chartWidget.getRefresh() > 0;
+                Integer refresh = chartWidget.getRefresh();
+                skipWidget = refresh != null && refresh > 0;
```

Comment:
chartWidget.getRefresh() returns an int so previous code was good and there is nothing to change.

Reply:
Yes, indeed. I alternated between making it an Integer or not when developing I believe, but sticked with int in the end.

Reply:
Resolved